catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/formulaeditor/FormulaEditorLogicListTest.java

```diff
@@ -0,0 +1,122 @@
+import org.catrobat.catroid.uiespresso.util.rules.BaseActivityInstrumentationRule;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import java.util.Arrays;
+
+import static android.support.test.espresso.Espresso.onData;
+import static android.support.test.espresso.Espresso.onView;
+import static android.support.test.espresso.action.ViewActions.click;
+import static android.support.test.espresso.matcher.ViewMatchers.withId;
+import static android.support.test.espresso.matcher.ViewMatchers.withText;
+import static android.support.test.espresso.assertion.ViewAssertions.matches;
+
+import static org.catrobat.catroid.uiespresso.content.brick.utils.BrickDataInteractionWrapper.onBrickAtPosition;
+import static org.hamcrest.Matchers.allOf;
+import static org.hamcrest.Matchers.instanceOf;
+import static org.hamcrest.core.Is.is;
+import static org.junit.runners.Parameterized.Parameter;
+import static org.junit.runners.Parameterized.Parameters;
+
+@Category({Cat.CatrobatLanguage.class, Level.Smoke.class})
+@RunWith(Parameterized.class)
+public class FormulaEditorLogicListTest {
+	@Rule
+	public BaseActivityInstrumentationRule<ScriptActivity> baseActivityTestRule = new
+			BaseActivityInstrumentationRule<>(ScriptActivity.class, true, false);
+
+	@Parameters(name = "{1}" + "-Test")
+	public static Iterable<Object[]> data() {
+		return Arrays.asList(new Object[][] {
+				{R.string.formula_editor_logic_and, "and"},
+				{R.string.formula_editor_logic_or, "or"},
+				{R.string.formula_editor_logic_not, "not"},
+				{R.string.formula_editor_function_true, "true"},
+				{R.string.formula_editor_function_false, "false"},
+				{R.string.formula_editor_logic_equal, "equal"},
+				{R.string.formula_editor_logic_notequal, "not_equal"},
+				{R.string.formula_editor_logic_lesserthan, "lesser_than"},
+				{R.string.formula_editor_logic_leserequal, "lesser_equal"},
+				{R.string.formula_editor_logic_greaterthan, "greater_than"},
+				{R.string.formula_editor_logic_greaterequal, "greater_equal"}
+		});
+	}
+
+	@Parameter
+	public @StringRes
+	int formulaEditorLogicFunction;
+
+	@Parameter(1)
+	public String testName;
+
+	@Before
+	public void setUp() throws Exception {
+		Script script = BrickTestUtils.createProjectAndGetStartScript("FormulaEditorLogicTest");
+		script.addBrick(new ChangeSizeByNBrick(0));
+		baseActivityTestRule.launchActivity(null);
+	}
+
+	@Test
+	public void testLogicListElements() {
+		onBrickAtPosition(1)
+				.onChildView(withId(R.id.brick_change_size_by_edit_text))
+				.perform(click());
+
+		onView(withText(R.string.formula_editor_logic))
+				.perform(click());
+		String editorFunction = UiTestUtils.getResourcesString(formulaEditorLogicFunction);
+
+		onData(allOf(is(instanceOf(String.class)), is(editorFunction)))
```

Comment:
it would be nice to have this encapsulated somewhere, since its also used in the above test class. also for ease of changing this whenever someone would refactor the list in the program so that it would have to be treated different, there would be only one place one has to touch.